packages/scrapers-v2/src/urls/scrape_urls.ts

```diff
@@ -0,0 +1,162 @@
+import { appendPath, loadHTML } from "../utils";
+import { CatalogHierarchy, College, CatalogPath } from "./types";
+
+/**
+ * Scrapes all catalog entries underneath the colleges.
+ * 
+ * @param start starting year (must be end year - 1)
+ * @param end   ending year
+ * @returns     a hierarchy of catalog entry links
+ */
+export const scrapeMajorLinks = async (start: number, end: number) => {
+  if (start !== end - 1) {
+    throw new Error("start should == end-1");
+  }
+
+  if (start < 2016) {
+    // this is because there is no HTML version of those catalogs
+    throw new Error("scraping for years before 2016-2017 are not supported");
+  }
+
+  if (start !== 2021) {
+    // todo: implement generic scraping from overall archive
+    // https://registrar.northeastern.edu/group/academic-catalogs/
+    throw new Error("only current year is supported");
+  }
+
+  return scrapeMajorLinksForUrl(
+    "https://catalog.northeastern.edu",
```

Comment:
seems like the base path can be a constant?

Reply:
yep we will clean this up when we get to this [ticket](https://trello.com/c/7aFTWtMU/452-scrapers-add-old-catalog-support)